nucypher/cli/painting.py

```diff
@@ -715,3 +715,109 @@ def write_deployed_allocations_to_csv(filepath: str, allocated: list, failed: li
             name = allocation.get('name', 'No name provided')
             row = (beneficiary, name, contract_address)
             writer.writerow(dict(zip(fieldnames, row)))
+
+
+def paint_worklock_status(emitter, registry: BaseContractRegistry):
+    from maya import MayaDT
+    worklock_agent = ContractAgency.get_agent(WorkLockAgent, registry=registry)
+    blockchain = worklock_agent.blockchain
+
+    # Agency
+    token_agent = ContractAgency.get_agent(NucypherTokenAgent, registry=registry)
+
+    # Time
+    start = MayaDT(worklock_agent.contract.functions.startBidDate().call())
+    end = MayaDT(worklock_agent.contract.functions.endBidDate().call())
+    duration = end - start
+    remaining = end - maya.now()
+
+    # TODO: Include calculated refund and deposit rates
+
+    payload = f"""
+
+Time
+======================================================
+Start Date ........ {start}
+End Date .......... {end}
+Duration .......... {duration}
+Time Remaining .... {remaining} 
+
+Economics
+======================================================            
+ETH Pool .......... {blockchain.client.get_balance(worklock_agent.contract_address)}
+Lot Size .......... {NU.from_nunits(worklock_agent.lot_value)} 
+Unclaimed Tokens .. {worklock_agent.get_unclaimed_tokens()}
+
+Boosting Refund ... {worklock_agent.contract.functions.boostingRefund().call()}
+Slowing Refund .... {worklock_agent.contract.functions.SLOWING_REFUND().call()}
+Refund Rate ....... {worklock_agent.get_refund_rate()}
+Deposit Rate ...... {worklock_agent.get_deposit_rate()}
+    """
+    emitter.message(payload)
+    return
+
+
+def paint_worklock_participant_status(emitter, registry, bidder_address):
+    WORKLOCK_AGENT = ContractAgency.get_agent(WorkLockAgent, registry=registry)
+
+    message = f"""
+WorkLock Participant {bidder_address}
+=====================================================
+Total Bid ............ {WORKLOCK_AGENT.get_bid(checksum_address=bidder_address)}
+Available Refund ..... {WORKLOCK_AGENT.available_refund(bidder_address=bidder_address)}
+Remaining Work ....... {WORKLOCK_AGENT.get_remaining_work(bidder_address=bidder_address)}
+"""
+    emitter.message(message)
+    return
+
+
+def paint_worklock_participant_notice(emitter, bidder_address, registry):
+    worklock_agent = ContractAgency.get_agent(WorkLockAgent, registry=registry)
+
+    obligation = f"""
+* WorkLock Participant Notice *
+-------------------------------
+
+- By participating in NuCypher's WorkLock you are committing to operating a staking NuCypher
+  node after the bidding window closes.
+
+- WorkLock token rewards are claimed in the form of a stake and will be locked for the stake duration.
+
+- WorkLock ETH deposits will be available for refund at a rate of {worklock_agent.economics.worklock_refund_rate} wei per confirmed period.
```

Comment:
Will be cool to add one more remark - that `refund_rate `will be freezed only after end of bidding

Reply:
:heavy_check_mark: